File: python/analysis.py
AGGRESSIVE_WORDS = [
    "idiot",
    "stupid",
    "hate",
    "kill",
    "fuck",
    "scam",
    "fraud",
    "chargeback",
    "refund",
    "paypal",
    "nitro",
    "steam",
    "crypto",
    "bastard",
    "cuxxl",
]


def analyze_priority(text: str, recent_tickets: int, repeated_reports: int):
    lower = text.lower()
    hits = [w for w in AGGRESSIVE_WORDS if w in lower]
    reasons = []
    if recent_tickets >= 3:
        reasons.append("High ticket volume in 24h")
    if repeated_reports >= 2:
        reasons.append("Repeated reports on same target")
    if hits:
        reasons.append(f"Flagged keywords: {', '.join(hits[:4])}")
    return {"priority": "HIGH" if reasons else "NORMAL", "reason": " - ".join(reasons) if reasons else None}


def find_aggressive_words(text: str):
    lower = text.lower()
    return [w for w in AGGRESSIVE_WORDS if w in lower]


SUGGESTIONS = [
    ("banned", "Provide ban-appeal steps and request context (username, reason, appeal notes)."),
    ("refund", "Ask for order ID, payment method, and transaction date."),
    ("chargeback", "Request evidence and explain chargeback policy."),
    ("scam", "Ask for screenshots, user IDs, and transaction links."),
]


def suggestions_from_text(text: str):
    lower = text.lower()
    return [s for k, s in SUGGESTIONS if k in lower]

File: src/tickets/analysis.ts
const AGGRESSIVE_WORDS = [
  'idiot',
  'stupid',
  'hate',
  'kill',
  'fuck',
  'scam',
  'fraud',
  'chargeback',
  'refund',
  'paypal',
  'nitro',
  'steam',
  'crypto',
  'bastard',
  'cuxxl',
  
];

export function analyzePriority(input: { text: string; recentTickets: number; repeatedReports: number }) {
  const text = input.text.toLowerCase();
  const hits = AGGRESSIVE_WORDS.filter((w) => text.includes(w));
  const reasons: string[] = [];

  if (input.recentTickets >= 3) reasons.push('High ticket volume in 24h');
  if (input.repeatedReports >= 2) reasons.push('Repeated reports on same target');
  if (hits.length) reasons.push(`Flagged keywords: ${hits.slice(0, 4).join(', ')}`);

  const priority = reasons.length ? 'HIGH' : 'NORMAL';
  return { priority, reason: reasons.join(' - ') || null };
}

const TEMPLATE_SUGGESTIONS: { keyword: string; suggestion: string }[] = [
  {
    keyword: 'banned',
    suggestion: 'Provide ban-appeal steps and request context (username, reason, appeal notes).',
  },
  {
    keyword: 'refund',
    suggestion: 'Ask for order ID, payment method, and transaction date.',
  },
  {
    keyword: 'chargeback',
    suggestion: 'Request evidence and explain chargeback policy.',
  },
  {
    keyword: 'scam',
    suggestion: 'Ask for screenshots, user IDs, and transaction links.',
  },
];

export function getSuggestionsFromText(text: string) {
  const lower = text.toLowerCase();
  return TEMPLATE_SUGGESTIONS.filter((t) => lower.includes(t.keyword)).map((t) => t.suggestion);
}


export function findAggressiveWords(text: string) {
  const lower = text.toLowerCase();
  return AGGRESSIVE_WORDS.filter((w) => lower.includes(w));
}
